tests/config_test.cc

```diff
@@ -0,0 +1,160 @@
+#include "cris/core/config/config.h"
+
+#include <gtest/gtest.h>
+
+#include <fmt/core.h>
+
+#include <chrono>
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+#include <thread>
+
+namespace cris::core {
+
+constexpr double kEps = 1e-6;
+
+namespace fs = std::filesystem;
+
+class ConfigTest : public testing::Test {
+   public:
+    ConfigTest()
+        : testing::Test()
+        , test_config_dir_(std::tmpnam(nullptr))
+        , test_config_path_(test_config_dir_ / "test.config") {
+        EXPECT_TRUE(fs::create_directory(test_config_dir_));
+    }
+
+    ~ConfigTest() { fs::remove_all(test_config_dir_); }
+
+    ConfigFile MakeConfigFile(std::string content) const {
+        std::ofstream config_file(test_config_path_);
+        config_file << std::move(content);
+        config_file.flush();
+        return ConfigFile(test_config_path_);
+    }
+
+   private:
+    fs::path test_config_dir_;
+    fs::path test_config_path_;
+};
+
+TEST_F(ConfigTest, Configs) {
+    const std::string int_config_name     = "int_config";
+    int               int_config_value    = 123;
+    const std::string double_config_name  = "double_config";
+    double            double_config_value = 3.14;
+
+    const std::string int_config_with_default_1_name  = "int_config_wd_1";
```

Comment:
Long java-ish name is really hard to read...
When not familiar with the code, I've forgot about what/why I'm reading on the other end of it...

Reply:
Any suggestions then?

Reply:
- Rely on some easy-to-get context
- Abbrev.
- Alignment

In this particular case:
- int_key
- int_val
- fp_key/dec_key
- fp_val/dec_val
- def_key
- def_val

wd2 overlaps with int_key so I would just skip that.

Reply:
Abbrevs sometimes introduce more guessing.

`int_` and `fp_`s are fine, but `def_` always make me think of "define" first.

wd2 is a different case. It has a default value and also a configured value.

Reply:
Really? I can't recall seeing def for default in var name like this.

Reply:
Not needed anyways